lib/lightning/work_orders.ex

```diff
@@ -273,14 +278,16 @@ defmodule Lightning.WorkOrders do
           Step.t() | Ecto.UUID.t(),
           [work_order_option(), ...]
         ) ::
-          {:ok, Run.t()} | {:error, Ecto.Changeset.t(Run.t())}
-  def retry(run, step, opts)
+          {:ok, Run.t()} | {:error, Ecto.Changeset.t()}
+  def retry(%Run{id: run_id}, %Step{id: step_id}, opts) do
+    retry(run_id, step_id, opts)
+  end
 
   def retry(run_id, step_id, opts)
       when is_binary(run_id) and is_binary(step_id) do
-    attrs = Map.new(opts)
-
-    run =
+    Multi.new()
```

Comment:
`do_retry` already uses a Multi, this is just an anticipation to put all queries in the multi so that it can return on `input_dataclip_id` multi if the step dataclip has been wiped.